ibmcloud/SECURE_EXECUTION.md

```diff
@@ -0,0 +1,106 @@
+# Secure Execution Support for Peer Pods on IBM Cloud VPC
+
+IBM Cloud VPC already supports to create Hyper Protect Virtual Server on LinuxONE. And Confidential Computing is enabled by using the [IBM Secure Execution](https://www.ibm.com/docs/en/linux-on-systems?topic=execution-introduction) for Linux technology. For more information, please refer to [Confidential computing with LinuxONE](https://cloud.ibm.com/docs/vpc?topic=vpc-about-se)
+
+To support Secure Execution for Peer Pods on IBM Cloud VPC, we need build a Secure Execution enabled custom VM image for pod VMs, to replace the non-SE custom image. This document describes how to build a SE enabled custom image, based on the exsiting [image build script](./image/build.sh) for IBM Cloud VPC.
+
+## Set up a demo environment without Secure Execution
+
+Follow the [README.md](./README.md) to setup a demo environment on IBM Cloud VPC. We need create a **s390x** architecture based cluster, so please include the following two lines in the `terraform.tfvars` file:
+```
+instance_profile_name = "bz2-2x8"
+image_name = "ibm-ubuntu-18-04-1-minimal-s390x-3"
+```
+
+## Prepare worker node to build SE enabled image
+
+1. **Build genprotimg**
+
+`genprotimg` is used to build an encrypted boot record from a given kernel, initial RAM disk, parameters, and public host-key document.
+
+For Ubuntu 18.04, we can build it from the source on GitHub:
+```bash
+git clone https://github.com/ibm-s390-linux/s390-tools.git
+cd s390-tools/genprotimg/
+apt install -y libglib2.0-dev libssl-dev libcurl4-openssl-dev
+make install
+```
+
+2. **Donwload host key document from Resource Link**
+
+The host key must match the host system for which the image is prepared. Please refer to document [Obtaining a host key document from Resource Link](https://www.ibm.com/docs/en/linux-on-systems?topic=execution-obtain-host-key-document#lxse_obtain_hkd) for details.
+
+As a registered user, access the search page:
+```
+https://www.ibm.com/servers/resourcelink/hom03010.nsf/pages/HKDSearch?OpenDocument
+```
+
+Need input the **machine type** and **machine serial number**, which can be found in an IBM internal repository(https://github.ibm.com/ZaaS/iaas_configuration), according to current VPC region and zone. It would be the last 7 characters of the **CPCA** value.
```

Comment:
well.. external users cannot access this internal repository. I would rather remove it here.

If users have access to a VSI instance, they could obtain machine type and machine serial from `/proc/sysinfo`.

@stevenhorsman @magowan How to provide those VPC region/zone specific information in the cloud documentation?

Reply:
Thanks for the info. I will update document to obtain machine serial from `/proc/sysinfo` on s390x worker.

Reply:
You could also reference this documentation: https://www.ibm.com/docs/en/linux-on-systems?topic=tasks-find-machine-serial